### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by nijotz**
## CHANGE LOG
* Move to Github actions for CI

## INTERNAL LINKS
https://newrelic.atlassian.net/browse/NODE-2380

### CODE REVIEW

**Comment by nijotz** [comment]
I ran the tests on a fork and everything worked. It skipped private and versioned tests: https://github.com/nijotz/node-newrelic/actions/runs/108657480

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
+name: Node Agent CI
+
+on: [push, pull_request]
```
Comment (michaelgoin) [inline .github/workflows/ci-workflow.yml:3]: I'm noticing the branch + PR plus matrix jobs gets pretty noisy. Not a big deal but we can consider limiting to certain branches in the future.

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    - name: Install Dependencies
+      run: npm ci
+    - name: Run Docker Services
+      run: npm run services
+    - name: Run Versioned Tests
+      run: npm run secrets
+      env:
+        LASP_LICENSE: ${{ secrets.LASP_LICENSE }}
+        LASP_SECURE_LICENSE: ${{ secrets.LASP_SECURE_LICENSE }}
+        BENDER_LICENSE: ${{ secrets.BENDER_LICENSE }}
```
Comment (michaelgoin) [inline .github/workflows/ci-workflow.yml:103]: Would prob make agnostic in naming here. GENERIC_LICENSE or TEST_LICENSE or similar.

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+  versioned:
+    runs-on: ubuntu-latest
+
+    if: github.repository == 'newrelic/node-newrelic'
```
Comment (michaelgoin) [inline .github/workflows/ci-workflow.yml:110]: I see what aren't running versioned tests for forked repos. Likely a necessary short-term step but another spot to ideally resolve related to ongoing discussions.

File: .github/workflows/ci-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      run: npm ci
+    - name: Run Docker Services
+      run: npm run services
+    - name: Run Versioned Tests
+      run: npm run secrets
+      env:
+        LASP_LICENSE: ${{ secrets.LASP_LICENSE }}
+        LASP_SECURE_LICENSE: ${{ secrets.LASP_SECURE_LICENSE }}
+        BENDER_LICENSE: ${{ secrets.BENDER_LICENSE }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (michaelgoin) [inline .github/workflows/ci-workflow.yml:104]: I don't believe these two AWS secrets are needed here.

**Pull request merged**